components/server/src/code-sync/code-sync-service.ts

```diff
@@ -303,4 +292,26 @@ export class CodeSyncService {
         return router;
     }
 
+    protected async deleteObjects(userId: string, resourceKey: ServerResource, revs: string[]) {
+        const tasks = revs.map(rev => this.db.deleteResource(userId, resourceKey, rev, async (rev: string) => {
+            const obj = toObjectName(resourceKey, rev);
+            try {
+                const request = new DeleteRequest();
+                request.setOwnerId(userId);
+                request.setExact(obj);
+                const blobsClient = this.blobsProvider.getDefault();
+                await util.promisify<DeleteRequest, DeleteResponse>(blobsClient.delete.bind(blobsClient))(request);
+            } catch (err) {
+                if (err.code === status.NOT_FOUND) {
+                    return;
+                }
+                log.error({ userId }, 'code sync: failed to delete obj', err, { object: obj });
+                throw err;
+            }
+        }).catch(err => {
+            log.error({ userId }, 'code sync: failed to delete', err);
+        }))
+        return Promise.all(tasks)
```

Comment:
> Any of them failed is ok, we should continue to delete other items

Then is better to use `Promise.allSettled`

Reply:
👍 Great! I have never used this func before! but in this case, we need some logs which may relate to DB right? We can use `Promise.allSettled` to make it more readable.

Reply:
🙏 updated @jeanp413

Reply:
If we keep the catch here then `Promise.all(tasks) will always resolve successfully and no need to change it to `Promise.allSettled`, other option is to remove catch here and use `Promise.allSettled` handle any error in caller function

Reply:
I was thinking of something like this, but it's fine either way, feel free to resolve this
```typescript
this.deleteObjects(userId, resourceKey, oldRevList).then(results => {
	results.forEach(p => {
		if (p.status === 'rejected') {
			log.error({ userId }, 'code sync: failed to delete', err);
		} 
	});
});
```